### DISCUSSION TITLE
All of a sudden self hosted Renovate runs are taking a lot longer.

### DISCUSSION THREAD

**Discussion by moltar** [renovatebot/renovate#27916] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

renovatebot/github-action@v39.2.1

### Please tell us more about your question or problem

Using the GH action `renovatebot/github-action@v39.2.1` for runnign Renovate.

By default it always picks up the latest version for each run.

Until about March 11 6:20 UTC, our runs were always in the range of 6-7 minutes.

After that time, all of the runs are taking > 10 minutes and are timing out (as that's our GitHub workflow limit).

Are there any changes in Renovate that could've contributed to these prolonged runs?

We haven't changed anything else on our end.

- Did not add or remove any deps during this time
- Did not add or remove any code

The only other changes that have occurred were Renovate upgrade PRs themselves.

Could also be GH runners, maybe GH replaced with some underpowered machines?

Thanks.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Are you using repositoryCache?

**Reply by rarkins** [reply]
Please also work out which version of renovate CLI is in use

**Comment by moltar** [comment]
> Are you using repositoryCache?

Not explicitly. Unless it is something that is enabled by default.

> Please also work out which version of renovate CLI is in use

The last good run was on `37.235.1`.

The first observed failing one was on`37.235.5`.

The latest failing one was on `37.241.0`.

**Reply by rarkins** [reply]
We are running 37.238.1 in the mend renovate app on github without any obvious performance decrease. Sometimes changes in code can result in cache invalidations but usually it recovers within one full "run" of repos

**Reply by moltar** [reply]
Thanks for confirming. Will look further then.

**Reply by rarkins** [reply]
Please try upgrading to latest and let me know if the performance degradation persists

**Reply by moltar** [reply]
Latest action or latest image?

Because the image is always latest, it's using the `latest` tag.

**Reply by viceice** [reply]
you need to ensure your runner always pulls the image and doesn't use a cached one
